cmd/ledger/main.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/hyperledger/fabric/internal/ledger"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app = kingpin.New("ledger", "Ledger Utility Tool")
+
+	compare       = app.Command("compare", "Compare two ledgers via their snapshots.")
+	snapshotPath1 = compare.Arg("snapshotPath1", "File path to first ledger snapshot.").Required().String()
+	snapshotPath2 = compare.Arg("snapshotPath2", "File path to second ledger snapshot").Required().String()
+
+	troubleshoot = app.Command("troubleshoot", "Identify potentially divergent transactions.")
+
+	args = os.Args[1:]
+)
+
+func main() {
+	kingpin.Version("0.0.1")
+
+	command, err := app.Parse(args)
+	if err != nil {
+		kingpin.Fatalf("parsing arguments: %s. Try --help", err)
+		return
+	}
+
+	switch command {
+
+	case compare.FullCommand():
+
+		err := ledger.Compare(*snapshotPath1, *snapshotPath2, "./result.json")
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		fmt.Println("Successfully compared snapshots")
```

Comment:
We'll provide this message whether there were any record differences or not.  I had a record difference, but received this message, and therefore thought the comparison resulted in no differences.  When I actually opened the result.json however, I see that there was one difference identified.

What I would suggest, is count the number of differences and report it here. And if there is one or more differences reported, print the JSON file name where the details can be viewed.  If there are no differences, then we don't even need to create the JSON file.

Reply:
That makes a lot of sense to me. I think adding a count would make it clear that "Successfully compared snapshots" means that there were no errors, not that there were no differences found.